app/models/manageiq/providers/autosde/storage_manager/ems_refresh_workflow.rb

```diff
@@ -1,15 +1,23 @@
 class ManageIQ::Providers::Autosde::StorageManager::EmsRefreshWorkflow < ManageIQ::Providers::EmsRefreshWorkflow
+  class << self
+    def start_refresh(target_class, target_id, ems_id, task_id, options = {})
+      options.reverse_merge!(:target_class => target_class, :target_id => target_id, :ems_id => ems_id, :native_task_id => task_id)
+      ManageIQ::Providers::Autosde::StorageManager::EmsRefreshWorkflow.create_job(options).tap { |job| job.signal(:start) }
+    end
+  end
 
   def run_native_op
     queue_signal(:poll_native_task)
   end
   alias start run_native_op
 
   def poll_native_task
-    case autosde_client.JobApi.jobs_pk_get(options[:native_task_id]).status
+    native_object = autosde_client.JobApi.jobs_pk_get(options[:native_task_id])
+    case native_object.status
     when "FAILURE"
       signal(:abort, "Task failed")
     when "SUCCESS"
+      # options[:native_object_id] = native_object.object_id
```

Comment:
> But for some reason in the refresh method it doesn't identify the change, and I don't understand why.

If this is the line you are talking about the issue is you have to save the changes to the database, right here they are only modified in memory.

```suggestion
      options[:native_object_id] = native_object.object_id
      save!
```

Reply:
Yes! that's exactly what I was talking about, and now you made me understand that for every refresh it looks for the job in the database and that it's not kept in memory.

So I squashed and tried it, it works well.

Do you suggest any other change? (or about the time interval...)